fedora-33/34 build fail on packaging ARM

There are unexpected fails during the packaging step on ARM64 on fedora-33/34
Tarantool version: tarantool-2.9.0.367

```
/build/usr/src/debug/tarantool-2.9.0.367/third_party/curl/lib/transfer.c:56:2: error: #error "We can't compile without socket() support!"
   56 | #error "We can't compile without socket() support!"
      |  ^~~~~
[ 75%] Building C object lib/CMakeFiles/libcurl.dir/version_win32.c.o
make[6]: *** [lib/CMakeFiles/libcurl.dir/build.make:1630: lib/CMakeFiles/libcurl.dir/transfer.c.o] Error 1
make[6]: *** Waiting for unfinished jobs....
/build/usr/src/debug/tarantool-2.9.0.367/third_party/curl/lib/url.c:57:2: error: #error "We can't compile without socket() support!"
   57 | #error "We can't compile without socket() support!"
      |  ^~~~~
```
This is a [failed worklow](https://github.com/tarantool/tarantool/runs/3379706353?check_suite_focus=true)
The passed workflow you can find [here](https://github.com/tarantool/tarantool/runs/3351171483?check_suite_focus=true)

--- a/cmake/BuildLibCURL.cmake
+++ b/cmake/BuildLibCURL.cmake
@@ -144,6 +144,11 @@ macro(curl_build)
     list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_DISABLE_IMAP=ON")
     list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_DISABLE_SMTP=OFF")
 
+    # We need PIC at least to enable build for Fedora on
+    # ARM64 CPU. Without it configuration with Fedora
+    # link hardening fails.
+    list(APPEND LIBCURL_CMAKE_FLAGS "-DCMAKE_POSITION_INDEPENDENT_CODE=ON")
+
     include(ExternalProject)
     ExternalProject_Add(
         bundled-libcurl-project

